client/landing/stepper/declarative-flow/internals/hooks/use-sign-up-start-tracking/index.tsx

```diff
@@ -41,6 +42,16 @@ export const useSignUpStartTracking = ( { flow, currentStepRoute }: Props ) => {
 		}
 	}, [ queryParams, setQuery ] );
 
+	// recordSignupStart should always be logged on page load.
+	// this ensures it gets logged after a page refresh.
+	useEffect( () => {
+		if ( isInitialMount ) {
+			recordSignupStart( flowName, ref, extraProps || {} );
```

Comment:
This leads to the same situation we are trying to resolve in discussion https://github.com/Automattic/wp-calypso/issues/94175#issuecomment-2362724556

We shouldn't ignore any additional/extra props. If the event gets recorded multiple times because of it, then so be it. Analytics can pick the first. This is my understanding from it, and how we can proceed.